### PR TITLE
Respect the user option RAJA_ENABLE_EXTERNAL_CUB

### LINKED ISSUE #1194: RAJA does not respect my wish to not use an external CUB
https://github.com/LLNL/RAJA/blob/4a07809dfd0d8dcd4daf5eeb91fb692394c5815b/cmake/SetupPackages.cmake#L32

I don't want to use an external cub, but  no matter what I do, RAJA always uses the system install. I can't figure out if its even possible to use the submodule if there is a the is findable by `FindPackage`. Any advice?

### PR DESCRIPTION
**Pull request by mclarsen**
resolves #1194

# Summary

- This PR is a bugfix
- It does the following (modify list as needed):
  - Fixes #1194

### CODE REVIEW

File: cmake/SetupPackages.cmake [hunk 1]
```diff
@@ -29,7 +29,7 @@ if (RAJA_ENABLE_TBB)
   endif()
 endif ()
 
-if (RAJA_ENABLE_CUDA OR RAJA_ENABLE_EXTERNAL_CUB)
+if (RAJA_ENABLE_CUDA AND RAJA_ENABLE_EXTERNAL_CUB)
```
Comment (rhornung67) [inline cmake/SetupPackages.cmake:32]: We need to check that this will still pick up the submodule if that's what one wants.

Comment (mclarsen) [reply cmake/SetupPackages.cmake:32]: For me it worked. Before it wasn't installing the cub files and now it is.

Comment (rhornung67) [reply cmake/SetupPackages.cmake:32]: Great. I'm testing locally now. I added the folks who touched that file last for review. we'll discuss this at the RAJA meeting next Tuesday (Monday is a lab holiday).

**Comment by MrBurmark** [comment]
@mclarsen If #1209 fixes your issue, can you close this?

**Pull request closed without merging**